components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/extension/eidas/EidasExtensionProcessor.java

```diff
@@ -0,0 +1,354 @@
+                .equals(EidasConstants.EIDAS_PREFIX);
+    }
+
+    /**
+     * Process and Validate the SAML extensions in authentication request for EIDAS message format.
+     *
+     * @param request        Authentication request
+     * @param validationResp reqValidationResponseDTO
+     * @throws IdentityException
+     */
+    @Override
+    public void processSAMLExtensions(RequestAbstractType request, SAMLSSOReqValidationResponseDTO validationResp)
+            throws IdentitySAML2SSOException {
+
+        if (request instanceof AuthnRequest) {
+            if (log.isDebugEnabled()) {
+                log.debug("Process the extensions for EIDAS message format");
+            }
+            Extensions extensions = request.getExtensions();
+            if (extensions != null) {
+                validateForceAuthn(validationResp);
+                validateIsPassive(validationResp);
+                validateAuthnContextComparison(validationResp);
+                validateSPType(validationResp, extensions);
+
+                processRequestedAttributes(validationResp, extensions);
+            }
+        }
+    }
+
+    /**
+     * Process and Validate a response against the SAML request with extensions for EIDAS message format.
+     *
+     * @param response    SAML response
+     * @param assertion   SAML assertion
+     * @param authReqDTO Authentication request data object
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public void processSAMLExtensions(StatusResponseType response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO)
+            throws IdentitySAML2SSOException {
+
+        if (response instanceof Response) {
+            if (log.isDebugEnabled()) {
+                log.debug("Process the extensions for EIDAS message format");
+            }
+            validateMandatoryRequestedAttr((Response) response, assertion, authReqDTO);
+            setAuthnContextClassRef(assertion, authReqDTO);
+        }
+    }
+
+    private void validateMandatoryRequestedAttr(Response response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO) {
+
+        List<String> mandatoryClaims = getMandatoryAttributes(authReqDTO);
+        boolean isMandatoryClaimPresent = validateMandatoryClaims(assertion, mandatoryClaims);
+        if (!isMandatoryClaimPresent) {
+            response.setStatus(SAMLSSOUtil.buildResponseStatus(SAMLSSOConstants.StatusCodes.IDENTITY_PROVIDER_ERROR,
+                    errorMsg));
+            if (CollectionUtils.isNotEmpty(assertion.getAttributeStatements())) {
+                assertion.getAttributeStatements().clear();
+            }
+
+            NameID nameId = new NameIDBuilder().buildObject();
+            nameId.setValue("NotAvailable");
+            nameId.setFormat(NameIdentifier.UNSPECIFIED);
+            assertion.getSubject().setNameID(nameId);
+            return;
+        }
+        setAttributeNameFormat(assertion.getAttributeStatements());
```

Comment:
Better to add a debug log on which mandatory claims are being validated for which SP

Reply:
Added